File: test_go/src/pattern/factory/simple_factory/pizza_store.go
package simple_factory

//简单工厂，通过一个单独的类来创建对象
type SimplePizzaFactory struct {
}

func (this *SimplePizzaFactory) CreatePizza(tp string) Pizza {
	var pizza Pizza = nil
	if tp == "cheese" {
		pizza = NewCheesePizza()
	} else if tp == "pepperoni" {
		pizza = NewPepperoniPizza()
	} else if tp == "clam" {
		pizza = NewClamPizza()
	} else if tp == "veggie" {
		pizza = NewVeggiePizza()
	}
	return pizza
}

/*--------------------------------------------------------------------------------------------------------------*/
type PizzaStore struct {
	factory *SimplePizzaFactory
}

func NewPizzaStore(factory *SimplePizzaFactory) *PizzaStore {
	this := new(PizzaStore)
	this.factory = factory
	return this
}

func (this *PizzaStore) OrderPizza(tp string) Pizza {
	pizza := this.factory.CreatePizza(tp)

	pizza.Prepare()
	pizza.Bake()
	pizza.Cut()
	pizza.Box()
	return pizza
}

File: java_test/src/DesignPattern/Factory/FactoryMethod/PizzaStore.java
package DesignPattern.Factory.FactoryMethod;

/**
 * Created by Administrator on 2018/4/4.
 */
public abstract class PizzaStore {
    abstract Pizza createPizza(String item);

    public Pizza orderPizza(String type) {
        Pizza pizza = createPizza(type);

        System.out.println("--- Making a " + pizza.getName() + " ---");
        pizza.prepare();
        pizza.bake();
        pizza.cut();
        pizza.box();
        return pizza;
    }
}
